lib/compiler/passes/generate-bytecode.js

```diff
@@ -244,6 +247,8 @@ function generateBytecode(ast) {
   }
 
   function buildSimplePredicate(expression, negative, context) {
+    const match = expression.match | 0;
```

Comment:
what is `expression.match`?  where does it come from?  what does it do?  why is it named this, when none of the things in the language names "match" are a number?  what's with the bitwise or of zero?

Reply:
> that guaranties, that the result is integer with value 0, and not undefined if that property doesn't exist

this is absurdly over-tricky code for the desired result.  you are attempting to be an optimizing compiler, and expecting javascript programmers to know your intent, these consequences, and incorrectly believing that transcompilers won't replace this with things without these guarantees.

the desired approach is

```typescript
if (somevariable === 0)
```

This will ensure that the result is a number (javascript doesn't have integers) with value `0`, and not `undefined`

It's also faster, conceptually describes correctly what it does, and doesn't require a trivia course to infer intent

Reply:
> This is semi-standard way to say the JS engine that there expected an integer. 

No it isn't.

Stop resolving things without the commenter's permission.  This isn't resolved.

&nbsp;

> Even a whole library grew out of this -- ASM.js

This is not asm.js, this is javascript.  This is why I said "you are attempting to be an optimizing compiler."

I already pointed out a bug this causes.  You continued to explain, instead of understanding the bug.

Reply:
>  undefined (== absence of the property) should be treated like the value 0. Your simple check fails on that

This is not correct.